google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/module/JavaLanguageLevelSupportCheck.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.module;
+
+import com.google.cloud.tools.intellij.appengine.cloud.AppEngineEnvironment;
+import com.google.cloud.tools.intellij.appengine.facet.AppEngineFacet;
+import com.google.cloud.tools.intellij.appengine.project.AppEngineAssetProvider;
+import com.google.cloud.tools.intellij.appengine.project.AppEngineProjectService;
+import com.google.cloud.tools.intellij.util.GctBundle;
+
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationDisplayType;
+import com.intellij.notification.NotificationGroup;
+import com.intellij.notification.NotificationListener;
+import com.intellij.notification.NotificationType;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.module.EffectiveLanguageLevelUtil;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleComponent;
+import com.intellij.openapi.roots.LanguageLevelModuleExtension;
+import com.intellij.openapi.roots.ModifiableRootModel;
+import com.intellij.openapi.roots.ModuleRootManager;
+import com.intellij.pom.java.LanguageLevel;
+import com.intellij.psi.xml.XmlFile;
+import com.intellij.psi.xml.XmlTag;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+
+import javax.swing.event.HyperlinkEvent;
+
+public class JavaLanguageLevelSupportCheck implements ModuleComponent {
+
+  private static final String COMPONENT_NAME = "App Engine Java Version Support Check";
+  private static final String RUNTIME_TAG = "runtime";
+  private static final String RUNTIME_TAG_JAVA_8 = "java8";
+  private static final String UPDATE_HREF = "#update";
+  private static final LanguageLevel HIGHEST_SUPPORTED_LANGUAGE_LEVEL = LanguageLevel.JDK_1_7;
+
+  private Module thisModule;
+
+  public JavaLanguageLevelSupportCheck(Module module) {
+    this.thisModule = module;
+  }
+
+  @Override
+  public void moduleAdded() {
+    if (isModuleUsingUnsupportedLanguageLevel(thisModule)) {
+      warnUser(thisModule);
+    }
+  }
+
+  protected boolean isModuleUsingUnsupportedLanguageLevel(Module module) {
+    // if it's not an app engine module, it's fine
+    if (!hasAppEngineFacet(module)) {
```

Comment:
The way I was thinking about it, the reason that java 8 would be unsupported in a module is very much predicated on the module being an app engine module.  I will update the method name though, to make that more clear.